app/services/support_interface/tad_degree_class_export.rb

```diff
@@ -0,0 +1,91 @@
+module SupportInterface
+  class TADDegreeClassExport
+    def self.run_weekly
+      data_export = DataExport.create!(
+        name: 'Weekly export of the applications ministerial report',
+        export_type: :tad_degree_class,
+      )
+      DataExporter.perform_async(SupportInterface::TADDegreeClassExport, data_export.id)
+    end
+
+    def call(*)
+      report = choices_with_courses_and_subjects.find_each.map do |choice|
+        next if choice.phase == 'apply_2' && !choice.is_latest_a2_app
+
+        subject = MinisterialReport.determine_dominant_course_subject_for_report(choice.course_name, choice.course_level, choice.subject_names.zip(choice.subject_codes).to_h)
+
+        {
+          subject: subject,
+          route: choice.route,
+          degree_class: choice.degree_class,
+          applications: 0,
+          offers_received: 0,
+          number_of_acceptances: 0,
+          number_of_declined_applications: 0,
+          number_of_rejected_applications: 0,
+          number_of_withdrawn_applications: 0,
+          course_name: choice.course_name,
+          course_level: choice.course_level,
+          subject_names: choice.subject_names,
+          subject_codes: choice.subject_codes,
+          status: choice.status,
+        }
+      end
+
+      report = sort_and_group_by_subject_route_and_grade(report)
+
+      report.map do |key, ary|
+        ary.each do |hash|
+          MinisterialReport::DEGREE_CLASS_REPORT_STATUS_MAPPING[hash[:status].to_sym].each do |mapped_status|
+            report[key].first[mapped_status] += 1
+          end
+        end
+      end
+
+      report = report.values.flatten.each do |row|
+        row.delete(:course_name)
+        row.delete(:course_level)
+        row.delete(:subject_names)
+        row.delete(:subject_codes)
+        row.delete(:status)
+      end
+
+      report.reject! do |row|
+        row[:applications].zero? &&
+          row[:offers_received].zero? &&
+          row[:number_of_acceptances].zero? &&
+          row[:number_of_declined_applications].zero? &&
+          row[:number_of_rejected_applications].zero? &&
+          row[:number_of_withdrawn_applications].zero?
+      end
```

Comment:
I'm sure there must be a better way to do this 🤔

Reply:
If these were the only elements in the hash you could do `row.values.all?(&:zero?)` but I don't think you can avoid specifying the 6 keys so something like... `%i[applications offers_received number_of_acceptances number_of_declined_applications number_of_rejected_applications number_of_withdrawn_applications].all? { |key| row[key].zero? }`

But in a nutshell `Enum#any?` and `all?` are quite handy for this kind of thing.